portfolio/src/main/webapp/script.js

```diff
@@ -28,3 +28,24 @@ function addRandomFact() {
   // Add it to the page.
   document.getElementById('random-facts-container').innerText = randomFact;
 }
+
+function displayMessages() {
+  fetch('/data').then(response => response.json()).then((messageArray) => {
+    
+    const messageList = document.getElementById('message-container');
+    messageList.innerHTML = '';
+
+    for (var i = 0; i < messageArray.length; i++) {
+      messageList.appendChild(createListElement(messageArray[i]));
```

Comment:
messageList is a `<div>`, so you're adding `<li>` under that, which gives you
`<div>`
`  <li>...</li>`
`<div>`

That's not valid HTML. You should change the `<div>` to an `<ol>`
https://www.w3schools.com/tags/tryit.asp?filename=tryhtml_lists